### PR TITLE
Ecl kw metaclass

### PR DESCRIPTION
**Pull request by joakim-hove**

### CODE REVIEW

File: devel/python/python/ert/ecl/ecl_grid.py [hunk 1]
```diff
@@ -30,15 +30,79 @@
 import warnings
 import os.path
 import math
-from ert.cwrap import CClass, CFILE, CWrapper, CWrapperNameSpace
+from ert.cwrap import CFILE, BaseCClass
 from ert.util import IntVector
-from ert.ecl import EclTypeEnum, EclKW, ECL_LIB, FortIO
+from ert.ecl import EclPrototype, EclTypeEnum, EclKW, ECL_LIB, FortIO
```
Comment (jepebe) [inline devel/python/python/ert/ecl/ecl_grid.py]: ECL_LIB should be removed

File: devel/python/python/ert/ecl/ecl_rft.py [hunk 2]
```diff
@@ -19,12 +19,22 @@
 
 import types
 import warnings
-from ert.cwrap import CClass, CWrapper, CWrapperNameSpace
-from ert.ecl import EclRFTCell, EclPLTCell, ECL_LIB
+from ert.cwrap import BaseCClass, CClass, CWrapper, CWrapperNameSpace
```
Comment (jepebe) [inline devel/python/python/ert/ecl/ecl_rft.py]: CClass, CWrapper, CWrapperNameSpace, ECL_LIB should be removed

File: devel/python/python/ert/ecl/ecl_util.py [hunk 3]
```diff
@@ -25,8 +25,8 @@
 functions from ecl_util.c which are not bound to any class type.
 """
 import ctypes
-from ert.cwrap import create_enum, CWrapper, CWrapperNameSpace, BaseCEnum
-from ert.ecl import ECL_LIB
+from ert.cwrap import BaseCEnum
+from ert.ecl import EclPrototype, ECL_LIB
```
Comment (jepebe) [inline devel/python/python/ert/ecl/ecl_util.py]: ECL_LIB should be removed

File: devel/python/python/ert/ecl/fortio.py [hunk 4]
```diff
@@ -39,16 +39,31 @@
 import ctypes
 import os
 import sys
-from ert.cwrap import BaseCClass, CWrapper
-from ert.ecl import ECL_LIB
+from ert.cwrap import BaseCClass
+from ert.ecl import ECL_LIB, EclPrototype
```
Comment (jepebe) [inline devel/python/python/ert/ecl/fortio.py]: ECL_LIOB should be removed

File: devel/python/python/ert/geo/cpolyline_collection.py [hunk 5]
```diff
@@ -19,22 +19,35 @@
 import ctypes
 
 from ert.cwrap import BaseCClass, CWrapper
```
Comment (jepebe) [inline devel/python/python/ert/geo/cpolyline_collection.py]: Remove Wrapper

File: devel/python/python/ert/cwrap/basecclass.py [hunk 6]
```diff
@@ -109,3 +109,36 @@ def __del__(self):
                 # we can have a Python object with c_pointer == None.
                 if self.__c_pointer > 0:
                     self.free()
+
+                    
+        
+    def dropCPointer(self):
```
Comment (jepebe) [inline devel/python/python/ert/cwrap/basecclass.py]: I suggest prefixing with underscore. Could this be called `_invalidate` instead? Then the long comment is rather unnecessary...

File: devel/python/python/ert/ecl/ecl_rft.py [hunk 7]
```diff
@@ -19,12 +19,22 @@
 
 import types
 import warnings
-from ert.cwrap import CClass, CWrapper, CWrapperNameSpace
-from ert.ecl import EclRFTCell, EclPLTCell, ECL_LIB
+from ert.cwrap import BaseCClass
+from ert.ecl import EclRFTCell, EclPLTCell, ECL_LIB,EclPrototype
```
Comment (jepebe) [inline devel/python/python/ert/ecl/ecl_rft.py]: Forgotten ECL_LIB

**Comment by joakim-hove** [comment]
OK.

**Pull request merged**